ADR/0044-spdx-support.md

```diff
@@ -0,0 +1,302 @@
+                    - referenceCategory
+                    - referenceType
+                    - referenceLocator
+            - annotations
+                - `List<Annotation>`
+                    - annotationDate
+                    - annotationType
+                    - annotator
+                    - Comment
+    - relationships
+        - `List<Relationship>`
+            - spdxElementId
+            - relationshipType
+            - relatedSpdxElement
+
+#### 1:1 conversions
+Following CycloneDX to SPDX attributes are converted as 1:1 as they represent the same thing.
+
+| CycloneDX Attribute        | SPDX Attribute      |
+|----------------------------|---------------------|
+| components                 | packages            |
+| component.name             | package.name        |
+| component.version          | package.versionInfo |
+
+
+#### Component.purl
+CycloneDX (version 1.5) supports only a single purl attribute per component. SPDX doesn’t have a direct attribute, but instead every package includes an externalRefs array which describes all external references for the package. There are defined reference categories and types. For PURL, category PACKAGE-MANAGER and type purl is used. The purl itself will be stored as referenceLocator
+```
+| CycloneDX Attribute          | SPDX Attribute                                                |
+|------------------------------|---------------------------------------------------------------|
+| component.purl = `<PURL>`    | package.externalRefs = [{referenceCategory:”PACKAGE-MANAGER”, |
+|                              |                          referenceType:purl,                  |
+|                              |                          referenceLocator: `<PURL>`           |
+|                              |                          }]                                   |
+```
+
+#### Component.properties
+CycloneDX components properties describe mapping of string:string properties for given component. SPDX component doesn’t have anything similar to cyclonedx properties. SPDX Package annotations are the only attribute where custom data can be stored and the only “customizable” field where there is comment which is a simple string. Due to that fact, cycloneDX property in format of {“name”: <string>, “value”: <string>} is encoded into json string. There can be also annotations produced by other tools. Therefore to be able to tell annotation comment is json encoded, annotator should ends with string “:jsonencoded”. To indicate annotator was a tool, prefix “Tool:" has to be included in the field.
+
+```
+| CycloneDX Attribute                       | SPDX Attribute                                 |
+|-------------------------------------------|------------------------------------------------|
+| components.properties = [                 | package.annotations = [                        |
+|   {“name”: …, “value”: …}                 |   {..., annotator: "`Tool:<tool>`:jsonencoded” |
```

Comment:
```suggestion
|   {“name”: …, “value”: …}                 |   {..., annotator: "`Tool: <tool>`:jsonencoded” |
```

Reply:
Why would you want to have space there?

Reply:
I thought the missing space was a typo, based on the other mention of `Tool: <tool>` in this doc and https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#68-creator-field

Reply:
It looks like both variants are valid (according to validation tool) . I personally prefer one without space as when converting back to cyclonedx you can simply remove the prefix, but in the spec there is a space. So maybe lets put there a space just to be spec compatible.